### ISSUE TITLE
meson hardcoded paths..

### ISSUE THREAD

**Issue by 0verse** [mini-rose/coffee#18]
meson install does not honour --libdir setup since it is hard coded. It is perhaps better to let the users invoke meson with --perfix, --bindir, --libdir to install into preferred paths. There could be a default prefix, bindir, libdir etc. but meson.build should be overridden with user preferences when meson is invoked with "meson setup --prefix xxx --libdir xxx --bindir xxx" etc. Any ideas how to implement this?

**Comment by bellrise** [comment]
Yep, the hardcoded paths are now only default options, so when the user passes a different one like:

```
meson setup --bindir /usr/local/bin
```

It will put the nxg binary into there.

**Comment by 0verse** [comment]
well I use meson setup but it doesn't override. Anyway, the install_data() function you introduced results with build error on my build system:

> ../coffee-0.3~20230218/nxg/meson.build:46:0: ERROR: File /an/tutum/özge/.coffee-0.3~20230218/nxg does not exist

it builds if I do comment it like this:
```
#install_data(
#  exe.path(),
#  install_dir : '/usr/bin'
#)
```
isn't install_data used for "share" directory and not "bin"?

**Comment by bellrise** [comment]
It doesn't override because its only changed on a branch im working on now, its not on master.

**Comment by bellrise** [comment]
With what arguments would you call `meson setup` with?

**Comment by 0verse** [comment]
```
exec meson setup \
  --prefix         /an/duyum \
  --libdir         $veri \
  --libexecdir     $veri \
  --bindir         $etki \
  --sbindir        $etki \
  --includedir     $önek \
  --datadir        $veri \
  --mandir         $veri/man \
  --infodir        $veri/info \
  --localedir      $veri/locale \
  --sysconfdir     $ayar \
  --localstatedir  $duru \
  --sharedstatedir $duru/lib \
  --buildtype      plain \
  --auto-features  enabled \
  --wrap-mode      nodownload \
  -D               b_lto=true \
  -D               b_pie=true \
  "$@"
```

**Comment by bellrise** [comment]
Fixed as of e66e6cd

**Comment by 0verse** [comment]
well done bellrise :)
